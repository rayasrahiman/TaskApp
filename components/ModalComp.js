import React from 'react';
import {View, Text, Modal, Dimensions, StyleSheet} from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import IoniconS from 'react-native-vector-icons/Ionicons';

import Button from './Button';
import {Colors} from '../constants/colors';

export default function ModalComp({
  onRequestClose,
  iconName,
  title,
  subTitle,
  onPress,
  visible,
  btnTxt,
}) {
  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={visible}
      onRequestClose={onRequestClose}>
      <View style={styles.Container}>
        <View style={styles.modalView}>
          <IoniconS style={styles.checkMark} name={iconName} />
          <Text style={styles.modalTitleTxt}>{title}</Text>
          <Text style={styles.modalSubtTitleTxt}>{subTitle}</Text>
          <Button
            btnWrapper={styles.btnWrapper}
            title={btnTxt}
            onPress={onPress}
          />
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  Container: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
  },
  modalView: {
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 35,
    alignItems: 'center',
    shadowColor: Colors.black,
    width: Dimensions.get('window').width,
    height: hp(45),
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  checkMark: {
    color: Colors.green,
    fontSize: hp(10),
  },
  modalTitleTxt: {
    color: Colors.green,
    fontWeight: 'bold',
    fontSize: hp(4),
    textAlign: 'center',
    marginVertical: hp(2),
  },
  modalSubtTitleTxt: {
    color: Colors.grayishBlack,
    fontWeight: 'bold',
    fontSize: hp(3),
    textAlign: 'center',
    marginBottom: hp(3),
  },
  btnWrapper: {
    width: Dimensions.get('window').width - wp(10),
  },
});
