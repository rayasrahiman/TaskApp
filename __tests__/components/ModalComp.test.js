import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react-native';
import {shallow} from 'enzyme';

import ModalComp from '../../components/ModalComp';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('ModalComp', () => {
  test('should render ModalComp correctly', () => {
    const wrapper = render(<ModalComp />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
