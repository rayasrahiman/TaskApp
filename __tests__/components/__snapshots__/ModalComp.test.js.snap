// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComp should render ModalComp correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#244379",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 20,
          "elevation": 5,
          "height": 600.5,
          "padding": 35,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": 750,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#4D8405",
              "fontSize": 133.5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#4D8405",
            "fontSize": 53.5,
            "fontWeight": "bold",
            "marginVertical": 26.5,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#838383",
            "fontSize": 40,
            "fontWeight": "bold",
            "marginBottom": 40,
            "textAlign": "center",
          }
        }
      />
      <View
        style={
          Object {
            "marginHorizontal": 37.5,
            "width": 675,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#0E6E96",
              "borderRadius": 10,
              "height": 93.5,
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 40,
                "paddingVertical": 6.5,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
